rcl/include/rcl/dynamic_message_type_support.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__DYNAMIC_MESSAGE_TYPE_SUPPORT_H_
+#define RCL__DYNAMIC_MESSAGE_TYPE_SUPPORT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <rosidl_runtime_c/type_description/type_description__struct.h>  // TEMPORARY
+// #include <type_description_interfaces/msg/type_description.h>  // Use this when conversion is ok
+
+#include "rosidl_runtime_c/message_type_support_struct.h"
+#include "rcl/macros.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+
+// TODO(methylDragon): !!! Document that the user is in charge of the lifetime of the struct...
+// TODO(methylDragon): ...
+
+/// If the user passes a NULL desc, it is deferred instead, the middleware is responsibile for
+/// populating the fields on type discovery!!!
+///
+/// Does not take ownership of description (copies)
+/// Allocates the `ts` arg. The caller takes ownership of the `ts` arg.
```

Comment:
This one can be fixed in a follow up, but this docblock needs to be much more complete. Though it would be impacted by the allocator/init prs to follow up on these.